src/main/java/com/faforever/client/chat/emoticons/EmoticonService.java

```diff
@@ -0,0 +1,64 @@
+package com.faforever.client.chat.emoticons;
+
+import com.faforever.client.exception.ProgrammingError;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class EmoticonService implements InitializingBean {
+
+  private static final ClassPathResource EMOTICONS_JSON_FILE_RESOURCE = new ClassPathResource("images/emoticons/emoticons.json");
+
+  private final ObjectMapper objectMapper;
+
+  private List<EmoticonsGroup> emoticonsGroups;
+  private final HashMap<String, String> shortcodeToBase64SvgContent = new HashMap<>();
+  private Pattern emoticonShortcodeDetectorPattern;
+
+  @Override
+  public void afterPropertiesSet() throws IOException {
+    loadAndVerifyEmoticons();
+    initializePattern();
```

Comment:
I would move initializePattern into loadAndVerifyEmoticons so that way whenever emoticons are loaded the pattern is up to date